sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLViewSuite.scala

```diff
@@ -55,6 +54,76 @@ class SQLViewSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
     }
   }
 
+  test("error handling: existing a table with the duplicate name when creating/altering a view") {
+    withTable("tab1") {
+      sql("CREATE TABLE tab1 (id int)")
+      var e = intercept[AnalysisException] {
+        sql("CREATE OR REPLACE VIEW tab1 AS SELECT * FROM jt")
+      }.getMessage
+      assert(e.contains("The following is an existing table, not a view: `default`.`tab1`"))
+      e = intercept[AnalysisException] {
+        sql("CREATE VIEW tab1 AS SELECT * FROM jt")
```

Comment:
For this case, it shows `table already existed`. However, when we found a view/table exists, the error message is wrong:
```
View $tableIdentifier already exists. If you want to update the view definition,
 please use ALTER VIEW AS or CREATE OR REPLACE VIEW AS
```
We are unable to alter a table (that is not a view) by these commands `ALTER VIEW AS or CREATE OR REPLACE VIEW AS`.

See the [source code] (https://github.com/gatorsmile/spark/blob/c64092c6aaec42663278343a27467c1c8c165b92/sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala#L119-L124)